documentation/examples/custom-sd/adapter-usage/main.go

```diff
@@ -165,15 +169,23 @@ func (d *discovery) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 			continue
 		}
 
-		dec := json.NewDecoder(resp.Body)
-		err = dec.Decode(&srvs)
+		b, err := ioutil.ReadAll(resp.Body)
+		io.Copy(ioutil.Discard, resp.Body)
```

Comment:
thos should go in a defer, before the readall.

defer func() {
            io.Copy(ioutil.Discard, resp.Body)
            resp.Body.Close()
    }()

Reply:
The problem is, the requests are happening in a `for` loop every X seconds. So the only way to use `defer` in this case would be to factor out each individual HTTP request into its own function.